src/invidious/frontend/comments_youtube.cr

```diff
@@ -13,17 +13,17 @@ module Invidious::Frontend::Comments
           )
 
           replies_html = <<-END_HTML
-          <div id="replies" class="pure-g">
+          <div class="pure-g replies">
             <div class="pure-u-1-24"></div>
             <div class="pure-u-23-24">
               <p>
-                <a href="javascript:void(0)" data-continuation="#{child["replies"]["continuation"]}"
+                <a target="_blank" href="/comment_viewer?continuation=#{child["replies"]["continuation"]}&id=#{id}&type=#{type}" data-continuation="#{child["replies"]["continuation"]}"
```

Comment:
That `id` here is a potential XSS: when [`ucid` is retrieved [from `routes/channels.cr`](https://github.com/iv-org/invidious/blob/e6ca5b54ead6f8e2928992dbbcd21cb81c790be2/src/invidious/routes/channels.cr#L203), it's directly passed to the current function (`template_youtube()`, on line 233), without validation/encoding.

To fix that you can either use a regex or substitute characters inside `post(env)` (the route handler) or URL encode inside this function.